internal/controller/stas/containerimagescan_status.go

```diff
@@ -94,11 +104,18 @@ func (p *containerImageScanStatusPatch) apply(ctx context.Context, c client.Clie
 	var err error
 	// Repeat until resource fits in api-server by increasing minimum severity on failure.
 	for severity := *p.minSeverity; severity <= stasv1alpha1.MaxSeverity; severity++ {
-		p.patch.Status.Vulnerabilities, err = filterVulnerabilities(p.vulnerabilities, severity)
+		var vulnerabilities []stasv1alpha1.Vulnerability
+
+		vulnerabilities, err = filterVulnerabilities(p.vulnerabilities, severity)
 		if err != nil {
 			return err
 		}
```

Comment:
Could we somehow avoid the `var` and using same _outer err_ and _filterVulnerabilities_ err?

```suggestion
		vulnerabilities, vErr := filterVulnerabilities(p.vulnerabilities, severity)
		if vErr != nil {
			return vErr 
		}
```